sdk/resource/auto.go

```diff
@@ -53,7 +53,13 @@ func Detect(ctx context.Context, detectors ...Detector) (*Resource, error) {
 				continue
 			}
 		}
-		autoDetectedRes = Merge(autoDetectedRes, res)
+		autoDetectedRes, err = Merge(autoDetectedRes, res)
+		if err != nil {
+			errInfo = append(errInfo, err.Error())
+			if !errors.Is(err, ErrPartialResource) {
+				continue
+			}
```

Comment:
There's nothing after this in the loop.  Is the `continue` necessary?  Can `Merge()` even produce `ErrPartialResource`?

Reply:
Good catch. It is not needed and is just a result of careless copy/paste. Deleted.